### PR TITLE
JCRVLT-751 Bugfix/fix filename escaping in user export

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

**Comment by kwin** [comment]
@tmaret Please have a look

File: vault-core-it/vault-core-integration-tests/src/main/java/org/apache/jackrabbit/vault/packaging/integration/UserContentPackageIT.java [hunk 1]
```diff
@@ -105,6 +105,15 @@ public void installUserA_Replace_PkgPath() throws RepositoryException, IOExcepti
         installUserA(ImportMode.REPLACE, true, true);
     }
 
+    @Test
```
Comment (kwin) [inline vault-core-it/vault-core-integration-tests/src/main/java/org/apache/jackrabbit/vault/packaging/integration/UserContentPackageIT.java:108]: This test is IMHO wrong, as the to be tested package contains invalid filename encoding.

**Comment by tmaret** [comment]
Thank you @kwin ! For me either way is fine, we could escape the path in our code with PlatformNameFormat or have FV escape. I agree that we should aim for the option that is the most backward compatible.

**Comment by kwin** [comment]
@tmaret Maybe you can quickly check the AEM code base for usages of ExportOptions.setRootPath() to see how it was used?

**Comment by tmaret** [comment]
Good point

Adobe has only one user synchronisation feature that relies on `ExportOptions#setRootPath`. 

Apache Sling Content Distribution [also relies on that method](https://github.com/apache/sling-org-apache-sling-distribution-core/blob/08ae5df2b824c27acdc6b16f1b42a90efcd25f78/src/main/java/org/apache/sling/distribution/serialization/impl/vlt/VltUtils.java#L186).

Both modules would need to be patched which is manageable. IMO we should prioritise avoiding overall regression risk.

File: vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/ExportOptions.java [hunk 2]
```diff
@@ -102,15 +103,15 @@ public String getRootPath() {
      * generate packages that have a virtual root. If a root path different than '/' is set, the workspace filter
      * will be adjusted accordingly, if possible.
      *
-     * @param rootPath the root path
+     * @param rootPath the root path in platform format (compare with {@link PlatformNameFormat})
```
[the remark this answers is not in the document]
Comment (reschke) [reply vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/ExportOptions.java]: I really have no idea. Maybe @tripodsan ?

[the remark this answers is not in the document]
Comment (kwin) [reply vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/ExportOptions.java]: Already clarified below: https://github.com/apache/jackrabbit-filevault/pull/336#issuecomment-2085643579

**Comment by kwin** [comment]
If there are no objections, I am gonna merge tomorrow.

**Pull request merged**